### PR TITLE
Added unary negation to Diagonal and corresponding test

### PR DESCRIPTION
**Pull request by drrobots**
Unary minus/negation did not exist for Diagonal.  Added method and corresponding unit test.

Minor Changes in:
- base/linalg/diagonal.jl - added unary minus (negation) for Diagonal type
- test/linalg/diagonal.jl - added unit test unary minus

### CODE REVIEW

**Comment by kshyatt** [comment]
Nice! I especially :+1: the inclusion of a relevant test.

**Comment by drrobots** [comment]
After attending JuliaCon, I figured I'd make my first commit, and figured you had enough work to do.

**Comment by kshyatt** [comment]
It's wonderful to see another person working on LinAlg. Here's to many commits to come!

File: test/linalg/diagonal.jl [hunk 1]
```diff
@@ -43,6 +43,10 @@ for relty in (Float32, Float64, BigFloat), elty in (relty, Complex{relty})
     end
 
     debug && println("Simple unary functions")
+    for op in (-,)
+        @test_approx_eq op(D) op(DM)
```
Comment (andreasnoack) [inline test/linalg/diagonal.jl]: I think this should be tested with `==` since the computation doesn't involve arithmetic.

Comment (carnaval) [inline test/linalg/diagonal.jl]: and even then, would you be checking that you correctly negated zeros ?

I'll show myself out

**Comment by drrobots** [comment]
Took suggestion from @andreasnoack and made test exact.  Also learning how to use github poorly.

**Comment by andreasnoack** [comment]
Thank you for the contribution.

**Pull request merged**